src/server/search/doc_index.cc

```diff
@@ -168,6 +176,17 @@ void ShardDocIndices::InitIndex(const OpArgs& op_args, std::string_view name,
       });
 }
 
+bool ShardDocIndices::DropIndex(string_view name) {
+  return indices_.erase(name) > 0;
+}
+
+vector<string> ShardDocIndices::GetIndexNames() const {
+  vector<string> names{};
```

Comment:
Add `names.reserve(indices_.size())`?

Reply:
Done, even though it usually won't contain more than a few elements😄